modules/phase_field/src/action/GrandPotentialKernelAction.C

```diff
@@ -65,8 +66,20 @@ GrandPotentialKernelAction::validParams()
   parameters.addParam<bool>(
       "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
   MultiMooseEnum anisotropy("true=1 false=0", "false");
+  parameters.addParam<bool>("mass_conservation", false, "Whether strict mass conservation formulation is used or not");
   parameters.addRequiredParam<MultiMooseEnum>(
       "anisotropic", anisotropy, "Whether or not each mobility is anisotropic");
+  parameters.addParam<std::vector<NonlinearVariableName>>(
+      "concentrations", "List of concentration variables for strict mass conservation");
+  parameters.addParam<std::vector<MaterialPropertyName>>(
+      "body_forces", "List of body forces coefficients for strict mass conservation formulation."
+                     "Place in same order as switching_function_names.");
+  parameters.addParam<std::vector<MaterialPropertyName>>(
+      "mat_reacts", "List of mat react coefficients for strict mass conservation formulation."
+                     "Place in same order as switching_function_names.");
+  parameters.addParam<MaterialPropertyName>(
+      "mobility_cw", "Lcw", "Name of mobility to be used with concentration-chempot coupling."
```

Comment:
Hopefully we can remove this an just create a material property for `-1` in the action itself